subsys/bootloader/image/CMakeLists.txt

```diff
@@ -4,59 +4,50 @@
 # SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
 #
 
+if (NOT IMAGE)
+  if (CONFIG_SECURE_BOOT)
+    if (CONFIG_BOOTLOADER_MCUBOOT)
+      set(mcuboot_CONFIG_FW_METADATA "y" CACHE STRING "Build with FW_METADATA for B0")
```

Comment:
Use kconfig fragment by appending to a list for this image to set this kconfig, instead of using CACHE.